src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -353,14 +356,14 @@ public ResponseEntity<byte[]> getLecturePdfAttachmentsMerged(@PathVariable Long
 
         authCheckService.checkHasAtLeastRoleForLectureElseThrow(Role.STUDENT, lecture, user);
 
-        List<AttachmentUnit> lectureAttachments = attachmentUnitRepository.findAllByLectureIdAndAttachmentTypeElseThrow(lectureId, AttachmentType.FILE);
-
-        List<String> attachmentLinks = lectureAttachments.stream()
+        List<AttachmentUnit> lectureAttachments = attachmentUnitRepository.findAllByLectureIdAndAttachmentTypeElseThrow(lectureId, AttachmentType.FILE).stream()
                 .filter(unit -> authCheckService.isAllowedToSeeLectureUnit(unit, user) && "pdf".equals(StringUtils.substringAfterLast(unit.getAttachment().getLink(), ".")))
-                .map(unit -> FilePathService.getAttachmentUnitFilePath()
-                        .resolve(Path.of(String.valueOf(unit.getId()), StringUtils.substringAfterLast(unit.getAttachment().getLink(), "/"))).toString())
                 .toList();
 
+        lectureUnitService.completeAllLectureUnits(lectureAttachments, user, true);
+        List<String> attachmentLinks = lectureAttachments.stream().map(unit -> FilePathService.getAttachmentUnitFilePath()
+                .resolve(Path.of(String.valueOf(unit.getId()), StringUtils.substringAfterLast(unit.getAttachment().getLink(), "/"))).toString()).toList();
```

Comment:
Not in scope but maybe you can fix this anyway here:
We should fetch the actual links and not reconstruct it. This is prone to errors if the path generation changes.